src/markers.ts

```diff
@@ -75,6 +77,21 @@ export class FileMarker extends BaseGeoLayer {
         this.generateId();
     }
 
+    get isFrontmatterMarker(): boolean {
+        return !this.fileLine;
+    }
+
+    get backgroundColor(): string {
+        let htmlElement = this.parseHtml();
```

Comment:
Can't the HTML element be retrieved directly from Leaflet? Parsing the HTML this way every time you want to get the background color seems hacky and slow :-/

Reply:
I agree that this is both hacky and slow. However, Leaflet offers no way (that I can see) to retrieve attributes like background color directly when the icon is a DivIcon.

![image](https://github.com/esm7/obsidian-map-view/assets/153950304/7b21d7e8-c934-4eee-8b32-7d8ea09c9f87)

I can only access the top-level div element via this.icon?.options?.html. If you can think of a better way to accomplish this, that would be great. The best solution I can think of is to cache the values in the FileMarker so that parsing only happens once. This would address the speed issue but not necessarily the hackiness.